shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/StopDetailsDepartures.kt

```diff
@@ -18,73 +18,29 @@ data class StopDetailsDepartures(val routes: List<PatternsByStop>) {
         filterAtTime: Instant
     ) : this(
         global.run {
-            val loading = schedules == null || predictions == null
-            val tripMapByHeadsign = tripMapByHeadsign(stops, schedules, predictions, filterAtTime)
-
             val allStopIds =
                 if (patternIdsByStop.containsKey(stop.id)) {
-                    setOf(stop.id)
+                    listOf(stop.id)
                 } else {
-                    stop.childStopIds.toSet()
-                }
-
-            val patternsByRoute =
-                allStopIds
-                    .flatMap { patternIdsByStop[it] ?: emptyList() }
-                    .map { patternId -> routePatterns.getValue(patternId) }
-                    .groupBy { routes.getValue(it.routeId) }
-
-            val touchedLines: MutableSet<String> = mutableSetOf()
-
-            val activeRelevantAlerts =
-                alerts?.alerts?.values?.filter {
-                    it.isActive(filterAtTime) && it.significance >= AlertSignificance.Minor
+                    stop.childStopIds.filter { global.stops.containsKey(it) }
```

Comment:
question (non-blocking): When would a child stop be missing from the global data? Why is this new check needed?

Reply:
The child stops that are missing from the global data are the entrances and navigation nodes.